app/celery/tasks.py

```diff
@@ -47,16 +47,20 @@ def zip_and_send_letter_pdfs(filenames_to_zip, upload_filename):
         )
     except ClientError:
         current_app.logger.exception('FTP app failed to download PDF from S3 bucket {}'.format(folder_date))
-        task_name = "update-letter-notifications-to-error"
+        self.retry(queue='process-ftp-tasks')
```

Comment:
Minor: in the API we do [a bit of analysis to see if it's worth retrying](https://github.com/alphagov/notifications-api/blob/3702d4eae824994bcb50cb4a89bb50c371c1b829/app/aws/s3.py#L30). I'm pretty sure it's not worth doing here, considering the low number of retries, and the low general volume of ZIP files to send.

What do you think?

Reply:
There are 2 places we can get a ClientError, in the first case we check if the zip file is already in zips_sent, we check if the ClientError has ['Error']['Code'] == 'NoSuchKey'. In the second case we are getting the PDF, in get_zip_of_letter_pdfs_from_s3, I'd argue that it is extremely rare that the file would not be found, since we just ran an operation from the API when collating the files to send. 
I'd prefer not to add more complexity by adding the FileNotFound check.

Reply:
Thanks for explaining. I hadn't noticed we already do a bit of checking, and agree with you that we shouldn't need any more.